services/common/src/main/java/com/linkedin/openhouse/common/stats/model/BaseEventModels.java

```diff
@@ -0,0 +1,46 @@
+package com.linkedin.openhouse.common.stats.model;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.experimental.SuperBuilder;
+
+/** Container for base model classes used in the dataset event hierarchy. */
+public final class BaseEventModels {
+
+  private BaseEventModels() {
+    // Utility class - no instantiation
+  }
+
+  /** Base class containing table identification metadata. */
+  @Data
+  @SuperBuilder
+  @NoArgsConstructor
+  @AllArgsConstructor
+  public static class BaseDataset {
+
+    /** Name of the database for the dataset */
+    @NonNull private String databaseName;
+
+    /** Name of the table for the dataset */
+    @NonNull private String tableName;
+
+    /** Name of the cluster (e.g., holdem/war) */
+    @NonNull private String clusterName;
```

Comment:
Let's add a String tableMetadataLocation;

This will be fully qualified path to the root table metadata json. Useful to distinguish unique in case the table is dropped and recreated with the same name.